tests/BenchmarkDotNet.IntegrationTests.ManualRunning/ExpectedBenchmarkResultsTests.cs

```diff
@@ -122,64 +123,102 @@ private void AssertZeroResults(Type benchmarkType, IConfig config)
         {
-            AssertDifferentSizedStructsResults(ManualConfig.CreateEmpty()
+            AssertNonZeroResults(benchmarkType, ManualConfig.CreateEmpty()
                 .AddJob(Job.Default
                     .WithRuntime(runtimeMoniker.GetRuntime())
                 ));
         }
 
         [TheoryEnvSpecific("Can only run Full .NET Framework and Mono tests from Framework host", EnvRequirement.FullFrameworkOnly)]
-        [InlineData(RuntimeMoniker.Net462)]
-        [InlineData(RuntimeMoniker.Mono)]
-        public void DifferentSizedStructsBenchmarksReportsNonZeroTimeAndZeroAllocated_Framework(RuntimeMoniker runtimeMoniker)
+        [MemberData(nameof(NonEmptyFrameworkData))]
+        public void NonEmptyBenchmarksReportsNonZeroTimeAndZeroAllocated_Framework(Type benchmarkType, RuntimeMoniker runtimeMoniker)
         {
-            AssertDifferentSizedStructsResults(ManualConfig.CreateEmpty()
+            AssertNonZeroResults(benchmarkType, ManualConfig.CreateEmpty()
                 .AddJob(Job.Default
                     .WithRuntime(runtimeMoniker.GetRuntime())
                 ));
         }
 
-        private void AssertDifferentSizedStructsResults(IConfig config)
+        private void AssertNonZeroResults(Type benchmarkType, IConfig config)
         {
-            var summary = CanExecute<DifferentSizedStructs>(config
+            var summary = CanExecute(benchmarkType, config
                 .WithSummaryStyle(SummaryStyle.Default.WithTimeUnit(TimeUnit.Nanosecond))
                 .AddDiagnoser(new MemoryDiagnoser(new MemoryDiagnoserConfig(false)))
             );
 
             var cpuResolution = RuntimeInformation.GetCpuInfo().MaxFrequency?.ToResolution() ?? FallbackCpuResolutionValue;
-            var threshold = Threshold.Create(ThresholdUnit.Nanoseconds, cpuResolution.Nanoseconds);
 
             foreach (var report in summary.Reports)
             {
                 var workloadMeasurements = report.AllMeasurements.Where(m => m.Is(IterationMode.Workload, IterationStage.Actual)).GetStatistics().WithoutOutliers();
                 var overheadMeasurements = report.AllMeasurements.Where(m => m.Is(IterationMode.Overhead, IterationStage.Actual)).GetStatistics().WithoutOutliers();
 
-                bool isZero = ZeroMeasurementHelper.CheckZeroMeasurementTwoSamples(workloadMeasurements, overheadMeasurements, threshold);
+                // We use the default threshold here rather than using cpu resolution,
+                // because modern cpus can execute multiple instructions per clock cycle,
+                // resulting in measurements greater than 0 but less than 1 clock cycle.
+                // (example: Intel Core i9-9880H CPU 2.30GHz reports 0.2852 ns for `_field++;`)
+                bool isZero = ZeroMeasurementHelper.CheckZeroMeasurementTwoSamples(workloadMeasurements, overheadMeasurements);
```

Comment:
1. The current implementation of `CheckZeroMeasurementTwoSamples` is a bad one. I wrote it in 2018 when I had poor knowledge of statistics. I have done [my homework](https://aakinshin.net/tags/statistics/)](https://aakinshin.net/tags/statistics/) since then and now I know a much better way to implement it. The problem with the current implementation is that it is not reliable and the used statistical test is underpowered: it can produce too many false-negative results (there is an actual difference between workload and overhead, but the test is always green). However, a better implementation requires a new version of Perfolizer, which means a huge upgrade in terms of used APIs. While this is a huge change, it makes total sense to include it in v0.14.0.
2. Speaking of the default threshold, I do not like the concept of "default" value in general. If we have some knowledge of the domain area, we should use it to increase accuracy. However, it is not clear how to properly use it and what we should do in the case of unknown CPU frequency. Therefore, switching to a reasonable default, which is adaptive to the samples is not a bad idea. However, I do not like the current default of 1%. It is just a random value without justification. We should reconsider the default.
3. Finally, the test procedure should be adjusted. Outliers should not be removed (a proper test will handle any deviations), and the number of iterations should be increased (the default logic does not produce enough iterations for reasonable comparison). All the code I wrote here is methodically wrong and should be fixed.

Next action: I upgrade Perfolizer to the latest version so that we get a more reliable statistical engine. As a pleasant bonus, I will finally be sure of my checks of other PRs with low-level changes (now I cannot properly validate them so I procrastinate the merge).